File: week-03_05/tp/tp_py/tp1/solution/even_odd.py
from mpi4py import MPI

# MPI_Init() is done at the import

comm = MPI.COMM_WORLD
rank = comm.Get_rank()
size = comm.Get_size()

if rank % 2 == 0:
    print(f'I am the even-ranked process my rank is {rank}')
else:
    print(f'I am the odd-ranked process my rank is {rank}')

# MPI_Finalize() is done at exit

File: week-03_05/tp/tp/tp1/solution/even_odd.f90
program even_odd
  USE MPI_F08
  implicit none

  integer :: rank,nb_procs

  call MPI_INIT()

  call MPI_COMM_RANK(MPI_COMM_WORLD,rank)
  call MPI_COMM_SIZE(MPI_COMM_WORLD,nb_procs)

  if (mod(rank,2) == 0) then
    print *,'I am the even-ranked process my rank is ',rank
  else
    print *,'I am the odd-ranked process my rank is ',rank
  end if

  call MPI_FINALIZE()

end program even_odd
